src/redis-benchmark.c

```diff
@@ -748,6 +750,15 @@ static client createClient(char *cmd, size_t len, client from, int thread_id) {
             (int)sdslen(config.input_dbnumstr),config.input_dbnumstr);
         c->prefix_pending++;
     }
+
+    /* If the readonly config flag is set preprend the READONLY command */
+    if (config.readonly) {
```

Comment:
am I wrong or the readonly command will be sent on master nodes connections? in case so, maybe we should only set it in case it is a replica node (thus no need to check the global config)

Reply:
Currently, it will be sent to all nodes. I just implemented it that way because it was the simplest and the pre-existing prefix commands all rely on the global config to determine whether to append the command. Currently the client doesn't really have a way to determine if the cluster_node is a replica or a primary.

Reply:
I updated this to check if `replicate` is not null for the client's cluster_node as that will ensure that READONLY is only sent to replicas.